src/utils/src/ExponentialBackoffRetryStrategy.c

```diff
@@ -281,12 +292,15 @@ STATUS getExponentialBackoffRetryStrategyWaitTime(PRetryStrategy pRetryStrategy,
     return retStatus;
 }
 
-STATUS getExponentialBackoffRetryCount(PRetryStrategy pRetryStrategy, PUINT32 pRetryCount) {
+STATUS getExponentialBackoffRetryCount(PKvsRetryStrategy pKvsRetryStrategy, PUINT32 pRetryCount) {
     ENTERS();
     STATUS retStatus = STATUS_SUCCESS;
     PExponentialBackoffRetryStrategyState pRetryState = NULL;
-    CHK(pRetryStrategy != NULL && pRetryCount != NULL, STATUS_NULL_ARG);
-    pRetryState = TO_EXPONENTIAL_BACKOFF_STATE(pRetryStrategy);
+
+    CHK(pKvsRetryStrategy != NULL && pRetryCount != NULL, STATUS_NULL_ARG);
+    CHK(pKvsRetryStrategy->retryStrategyType != KVS_RETRY_STRATEGY_EXPONENTIAL_BACKOFF_WAIT, STATUS_INVALID_ARG);
```

Comment:
Shouldnt this be: 
`CHK(pKvsRetryStrategy->retryStrategyType == KVS_RETRY_STRATEGY_EXPONENTIAL_BACKOFF_WAIT, STATUS_INVALID_ARG);`

If `pKvsRetryStrategy->retryStrategyType` is `KVS_RETRY_STRATEGY_EXPONENTIAL_BACKOFF_WAIT`, `CHK` call evaluates to `FALSE`, which means we will not jump to CleanUp directly.

Reply:
Correct. Fixed
Also added check for `getExponentialBackoffRetryCount` in unit test which was missing.